content/2. Ondersteunende-informatie/4. Techniek/Git/4. Git Commits.md

```diff
@@ -0,0 +1,66 @@
+
+2. Hier zie je welke bestanden zijn veranderd, toegevoegd of verwijderd. Met de checkboxes kan je ook selecteren of je bepaalde bestanden niet wil commiten. Dit kan bijvoorbeeld met een test- of tijdelijk bestand.
+3. Hier voer je een titel (in dit geval "Update page.tsx") en een (optionele) beschrijving toe. Zodra een titel is toegevoegd kan het gecommit worden.
+	1. In deze beschrijving kan je een uitleg geven over de veranderingen in de commit. Ook kunnen hierin andere mensen getagd worden indien ze ook hebben gewerkt aan de veranderingen in die commit (met `@gebruikersnaam`).
+
+### Stashen
+Een stash is de plek waar de aanpassingen kunnen worden opgeslagen die niet meegenomen moeten worden in de commit. Hiermee kan een branch schoon achtegelaten worden. Hiermee kan je wisselen van branch zonder te hoeven commiten maar wel de veranderingen lokaal nog te bewaren. 
+
+### Tags
+Git tags zijn markeringen die je kunt gebruiken om specifieke punten in de geschiedenis van een Git-repository vast te leggen. Ze worden vaak gebruikt om belangrijke momenten, zoals de release van een nieuwe versie van software, te markeren. In tegenstelling tot branches, die in de loop van de tijd kunnen veranderen, blijven tags altijd aan dezelfde commit gekoppeld. 
+
+Tags worden veel gebruikt om versies van software te markeren, zoals `v1.0`, `v2.1.3`, enz. Dit helpt om een specifieke staat van de code vast te leggen, vooral bij releases. Omdat tags statisch zijn, kunnen ze altijd worden gebruikt om terug te gaan naar een exacte versie van de code. Tags maken het eenvoudig om belangrijke commits terug te vinden zonder door de volledige commit geschiedenis te moeten zoeken.
+
+## Wat kan je met een commit?
+Door het maken van een commit heb je een snapshot gemaakt van de bestanden. Dit zorgt ervoor dat de versiegeschiedenis van jouw project bij te houden is over de tijd heen. 
+
+Verder is het mogelijk om op een later moment terug te vallen op een commit, de zogeheten `rollback`. Voor meer informatie over hoe dit in werking gaat, zie [Git revert](https://www.atlassian.com/git/tutorials/undoing-changes/git-revert).
+
+### Pushen en pullen
+Na het commiten van code kan er *gepusht* worden. Dit houdt in dat alle veranderingen van één of meerdere commits naar de remote repository worden gezet. 
+
+**Ter illustratie:** Een aanpassing kan gezien worden als een brief. Deze brief stop je in een envelop, de **commit**. Deze doe je in de brievenbus om door te sturen, de **push**. Het is ook mogelijk om meerdere brieven in één keer in de brievenbus te doen, net zoals bij pushen van commits.
+
+Voor andere mensen om deze veranderingen ook te krijgen op hun lokale repositories kan er *gepulled* worden. Door te pullen worden de veranderingen van een  branch opgehaald van de remote repository en toegevoegd aan jouw huidige lokale repository. Hierbij is het mogelijk om conflicten te krijgen. Deze moeten eerst opgelost worden voordat er verder gewerkt kan worden.
```

Comment:
Opmerking: dus deze mensen halen de post uit hun brievenbus; leuke vergelijking.

Is het nog handig ergens te vermelden dat het gebruikelijk is om eerst te pullen voordat je pushed, en waarom?